framework/src/dirackernels/DiracKernel.C

```diff
@@ -228,7 +231,8 @@ DiracKernel::addPointWithValidId(Point p, unsigned id)
   // safe, because all processors have the same value of we_found_it.
   if (!we_found_it)
   {
-    const Elem * elem = _dirac_kernel_info.findPoint(p, _mesh);
+    const Elem * elem = blockRestricted() ? _dirac_kernel_info.findPoint(p, _mesh, &blockIDs())
+                                          : _dirac_kernel_info.findPoint(p, _mesh);
```

Comment:
It might be good to add ` mooseDoOnce(mooseWarning("..."))` to tell users that the point is not found in specified blocks.

Reply:
I am not sure if we should do that. Here is a comment from `DiracKernelInfo.C`:
```
  // Note: The PointLocator object returns NULL when the Point is not
  // found within the Mesh.  This is not considered to be an error as
  // far as the DiracKernels are concerned: sometimes the Mesh moves
  // out from the Dirac point entirely and in that case the Point just
  // gets "deactivated".
```
So this predates me -- it has been historically ignored.